tests/Report/JunitReporterTest.php

```diff
@@ -0,0 +1,265 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Report;
+
+use PhpCsFixer\Report\JunitReporter;
+use PhpCsFixer\Report\ReportSummary;
+
+/**
+ * @author Boris Gorbylev <ekho@ekho.name>
+ *
+ * @internal
+ */
+final class JunitReporterTest extends \PHPUnit_Framework_TestCase
+{
+    /** @var JunitReporter */
+    private $reporter;
+
+    protected function setUp()
+    {
+        $this->reporter = new JunitReporter();
+    }
+
+    /**
+     * @covers PhpCsFixer\Report\JunitReporter::getFormat
+     */
+    public function testGetFormat()
+    {
+        $this->assertSame('junit', $this->reporter->getFormat());
+    }
+
+    public function testGenerateNoErrors()
+    {
+        $expectedXml = <<<'XML'
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuites>
+  <testsuite name="PHP CS Fixer report" tests="1" assertions="1" failures="0" errors="0">
+    <testcase name="All OK" assertions="1"/>
+  </testsuite>
+</testsuites>
+XML;
+
+        $actualXml = $this->reporter->generate(
+            ReportSummary::create()
+                ->setChanged(array())
+        );
+
+        $this->assertJunitXmlSchema($actualXml);
+        $this->assertXmlStringEqualsXmlString($expectedXml, $actualXml);
+    }
+
+    public function testGenerateSimple()
+    {
+        $expectedXml = <<<'XML'
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuites>
+  <testsuite name="PHP CS Fixer report" tests="1" assertions="1" failures="1" errors="0">
+    <testcase name="someFile" class="someFile" file="someFile.php" assertions="1">
```

Comment:
`failures` is a number of failed `testcase`s
it should be in the `testsuite` only